go/types/struct.go

```diff
@@ -41,6 +41,29 @@ func skipStruct(dec *valueDecoder) {
 	}
 }
 
+func isStructSameTypeForSure(dec *valueDecoder, t *Type) bool {
```

Comment:
if this isn't going to return immediately in the false case, it should document why (i.e. to completely consume the struct from `dec`)